lib/statsd.js

```diff
@@ -38,20 +51,40 @@ const Client = function (host, port, prefix, suffix, globalize, cacheDns, mock,
     let socket;
```

Comment:
Seems like `createSocket` is getting more complex. Could this be moved outside of the constructor and into a utility function?

Reply:
I agree, but I'm not going to touch in this PR right now given it's size (and time I have).  Higher on my own personal refactoring list would be to start using async and get out of some of the ugly nested callbacks, one of these days.  :)

Reply:
Sounds good to me - let me know if I can help with that :)